### ISSUE TITLE
Type checking for dataclasses

### ISSUE THREAD

**Issue by jsh9** [agronholm/typeguard#161]
Hi, I'm wondering if there can be support for type checking dataclasses?  Currently, when I put `@typeguard.typechecked` on top of a dataclass definition, no checking happens (unless I'm mistaken).

Thanks!

**Comment by svermeulen** [comment]
Here's what I'm doing for this case:

https://gist.github.com/svermeulen/8a6ad727113a6d07f78332152c1d33b2

**Comment by SamuelMarks** [comment]
@svermeulen Might give your gist a shot https://gist.github.com/svermeulen/8a6ad727113a6d07f78332152c1d33b2

I suppose avoiding maintaining my own runtime type checker is a good idea…

**Comment by SamuelMarks** [comment]
@svermeulen Might give your gist a shot

I suppose avoiding maintaining my own runtime type checker is a good idea…

**Comment by dblock** [comment]
@svermeulen Your gist works! Want to PR this to this project?

**Comment by svermeulen** [comment]
Feel free to PR it.  I've added MIT license now

**Comment by idvorkin** [comment]
Did this get merged?

**Comment by agronholm** [comment]
No, and I'm not sure what approach I want to take with regards to checking class/instance variable assignments. I don't think I will add support in v3.0. Maybe in a later release.

**Comment by faph** [comment]
That's a shame. With typeguard 2 this used to work if I am not mistaken.

**Comment by agronholm** [comment]
It never did work with typeguard 2.

**Comment by dkamm** [comment]
Hi @agronholm - are you planning to support this in typeguard 4?

**Comment by agronholm** [comment]
Maybe in some follow-up minor release, but not the 4.0 release that is imminent.

**Comment by NicolasNarvaez** [comment]
In the case when one doesn't know what approach use to a solution like this, the best is to be flexible and give a temporary solution that works for the simpler and most fundamental case, that we know most devs would use, than to not give an alternative at all. I'm sure most of use are doing almost the same.
What is the issue with the proposed solution?, why not just split the logic for class, and instances, and weirder cases?
